mtags/src/main/scala-3/scala/meta/internal/pc/completions/OverrideCompletions.scala

```diff
@@ -104,6 +113,150 @@ object OverrideCompletions:
       .toList
   end contribute
 
+  def implementAllFor(
+      indexedContext: IndexedContext,
+      search: SymbolSearch,
+      autoImports: AutoImportsGenerator,
+      config: PresentationCompilerConfig
+  )(
+      td: TypeDef
+  )(using Context): List[l.TextEdit] =
+    def calcIndent(
+        td: TypeDef,
+        decls: List[Symbol],
+        source: SourceFile,
+        text: String,
+        shouldCompleteBraces: Boolean
+    )(using Context): (String, String, String) =
+      // For `FooImpl` in the below, the necessaryIndent will be 2
+      // because there're 2 spaces before `class FooImpl`.
+      // ```scala
+      // |object X:
+      // |  class FooImpl extends Foo {
+      // |  }
+      // ```
+      val necessaryIndent = inferIndent(
+        source.lineToOffset(td.sourcePos.line),
+        text
+      )
+      // infer indent for implementations
+      // If there's declaration in the class/object, follow its indent.
+      // For example, numIndent will be 8, becuase there're 8 spaces before
+      // `override def foo: Int`
+      // ```scala
+      // |object X:
+      // |  class FooImpl extends Foo {
+      // |        override def foo: Int = 1
+      // |  }
+      // ```
+      val numIndent =
+        decls.headOption
+          .map { decl =>
+            inferIndent(source.lineToOffset(decl.sourcePos.line), text)
+          }
+          .getOrElse(necessaryIndent + DefaultIndent)
+      val indent = " " * numIndent
+      val lastIndent =
+        if (td.sourcePos.startLine == td.sourcePos.endLine) ||
+          shouldCompleteBraces
+        then "\n" + " " * necessaryIndent
+        else ""
+      (indent, indent, lastIndent)
+    end calcIndent
+
+    val overridables = td.tpe.abstractTermMembers.map(_.symbol)
+    val completionValues = overridables
+      .map(sym =>
+        toCompletionValue(
+          sym.denot,
+          0, // we don't care the position of each completion value from ImplementAll
+          td,
+          indexedContext,
+          search,
+          shouldMoveCursor = false,
+          config,
+          shouldAddOverrideKwd = true
+        )
+      )
+      .toList
+
+    val (edits, imports) = toEdits(
+      completionValues,
+      autoImports
+    )
+
+    if edits.isEmpty then Nil
+    else
+      val text = indexedContext.ctx.source.content.mkString
+
+      // A list of declarations in the class/object to implement
+      val decls = td.tpe.decls.toList
+        .filter(sym =>
+          !sym.isPrimaryConstructor &&
+            !sym.is(ParamAccessor) && // `num` of `class Foo(num: int)`
+            sym.span.exists
+        )
+        .sortBy(_.sourcePos.start)
+
+      val source = indexedContext.ctx.source
+
+      val shouldCompleteBraces = decls.isEmpty && hasBraces(text, td).isEmpty
+
+      val (startIndent, indent, lastIndent) =
```

Comment:
Isn't startIndent always the same as indent ?

Reply:
Yes, it's always the same, but I just assigned them to  different variables, because I feel it's easier to read.